schematics/ng-generate/angularjs-hybrid/index.ts

```diff
@@ -0,0 +1,328 @@
+
+function addDependencies(): Rule {
+    return (tree: Tree, context: SchematicContext): Tree => {
+        addDependency(tree, '@ux-aspects/ux-aspects-ng1', UX_ASPECTS_NG1_VERSION);
+        addDependency(tree, '@angular/upgrade', latestVersions.Angular);
+        addDependency(tree, 'angular', ANGULARJS_VERSION);
+        addDependency(tree, 'jquery', JQUERY_VERSION);
+        addDependency(tree, 'jquery-ui', JQUERY_UI_VERSION);
+        context.addTask(new NodePackageInstallTask());
+        return tree;
+    };
+}
+
+function addTemplateFiles(options: Schema): Rule {
+    return (tree: Tree, context: SchematicContext) => {
+        const workspace = getWorkspace(tree);
+        const project = getProjectFromWorkspace(workspace, options.project);
+
+        // Get the dir for the hybrid-setup.ts file
+        const appModulePath = getAppModulePath(tree, getProjectMainFile(project));
+        const dir = dirname(appModulePath);
+
+        // Create hybrid-setup.ts from the template and move it to the target dir
+        const templateContext = {
+            ...options,
+            router: hasRouter(tree, appModulePath, 'AppModule')
+        };
+        const templateSource = apply(url('./files'), [applyTemplates(templateContext), move(dir)]);
+        const rule = branchAndMerge(chain([mergeWith(templateSource)]));
+
+        return rule(tree, context);
+    };
+}
+
+function addScripts(options: Schema): Rule {
+    return (tree: Tree) => {
+        const workspace = getWorkspace(tree);
+        const project = getProjectFromWorkspace(workspace, options.project);
+
+        addScriptImports(
+            project,
+            'node_modules/jquery/dist/jquery.js',
+            'node_modules/jquery-ui/ui/version.js',
+            'node_modules/jquery-ui/ui/widget.js',
+            'node_modules/jquery-ui/ui/data.js',
+            'node_modules/jquery-ui/ui/ie.js',
+            'node_modules/jquery-ui/ui/scroll-parent.js',
+            'node_modules/jquery-ui/ui/position.js',
+            'node_modules/jquery-ui/ui/unique-id.js',
+            'node_modules/jquery-ui/ui/widgets/mouse.js',
+            'node_modules/jquery-ui/ui/widgets/sortable.js',
+            'node_modules/bootstrap/dist/js/bootstrap.js',
+            'node_modules/angular/angular.js',
+            'node_modules/@ux-aspects/ux-aspects-ng1/ng1/ux-aspects-ng1.js'
+        );
+
+        return updateWorkspace(workspace);
+    };
+}
+
+function addStyles(options: Schema): Rule {
+    return (tree: Tree, _context: SchematicContext) => {
+        const workspace = getWorkspace(tree);
+        const project = getProjectFromWorkspace(workspace, options.project);
+
+        addStyleImportsAfter(
+            project,
+            'node_modules/@ux-aspects/ux-aspects/styles/ux-aspects.css',
+            'node_modules/@ux-aspects/ux-aspects-ng1/styles/ux-aspects.css'
+        );
+
+        // use the UX Aspects no legacy styles instead of the default stylesheet
+        replaceImport(
```

Comment:
I would suggest doing this before the `addStyleImportsAfter` call and then you can safely add the ng1 stylesheet after `node_modules/@ux-aspects/ux-aspects/styles/ux-aspects-no-legacy.css`. This would cover the edge case where someone has manually switched to the "no-legacy" stylesheet - Jane has done this in at least one of her projects (although it's very unlikely that those projects will adopt schematics).